spec/src/main/asciidoc/servlet-spec-body.adoc

```diff
@@ -2703,6 +2703,18 @@ resource after its call.
 * Actions on a servlet, on groups of servlets,
 or static content by zero, one, or more filters in a specifiable order.
 
+Be aware that any filter that modifies the `ServletResponse` can potentially
+break a subsequent filter or the servlet in the `FilterChain`. For example, if
+the servlet supports range requests or sets a content length, a filter that
+modifies the response body is likely to break the servlet. It is the
+responsibility of the filter developer to clearly describe what it does with the
+`ServletResponse` so a user of the filter is aware of what happens when they add
+the filter.
+
+Filters that modify the response body generated by a Servlet should take steps
+to ensure that they minimise the risk of breaking a subsequent filter or the
+servlet in the `FilterChain`.
```

Comment:
Do we want to be explicit and say that when modifying the body, a filter must ensure that any affected headers e.g. `Content-Length` are correctly modified as well.

Reply:
That was where I started in #25 because a TCK filter was breaking Tomcat's default servlet. The argument in #25 was a filter doesn't know what a subsequent filter/servlet might do to the response.
I'm fine with being more explicit and requiring that headers are correctly modified. It will mean updating the TCK to fix the filters that don't currently do this (which in those cases I think just means clearing the content-length header). Unless there are objections, I'll add that to my TODO list.

Reply:
No objections. Updated accordingly.